docs/operator_context.md

```diff
@@ -0,0 +1,112 @@
+# Operator context
+
+This is a resource used to define values for communicating with admin api of
+tyk ce/pro deployment.
+
+`OperatorContext` can be referenced on all custom resources by `contextRef` property.
+When a custom resource is applied with `contextRef` set then all the operations
+conducted by the operator will use the `OperatorContext` supplied by the `contextRef` 
+to perform reconciliation.
+
+# Defining OperatorContext
+
+Annotated `OperatorContext` for a community edition deployment looks like this
+
+```yaml
+apiVersion: tyk.tyk.io/v1alpha1
+kind: OperatorContext
+metadata:
+  name: community-edition
+spec:
+  env:
+    # The mod e of the admin api
+    # ce - community edition (open source gateway)
+    # pro - dashboard (requires a license)
+    mode: ce
+    # The authorization token this will be set in x-tyk-authorization header on the
+    # client while talking to the admin api
+    auth: foo
+    # Org ID to use
+    org: myorg
+    # The url to the tyk open source gateway deployment admin api
+    url: http://tyk.tykce-control-plane.svc.cluster.local:8001
+    # Set this to true if you want to skip tls certificate and host name verification
+    # this should only be used in testing
+    insecureSkipVerify: true
+    # For ingress the operator creates and manages ApiDefinition resources, use this to configure
+    # which ports the ApiDefinition resources managed by the ingress controller binds to.
+    # Use this to override default ingress http and https port
+    ingress:
+      httpPort: 8000
+      httpsPort: 8443
+```
+
+# Using secret for sensitive information
+
+It is not recommended to set `.spec.env.auth` in the custom resource `yaml` because it is
+a sensitive data that third party can use to access your tyk deployment. You
+should use secret instead.
+
+Create a `k8s` secret `tyk-operator-system` with our sensitive info for auth and values
+
+```sh
+kubectl create secret -n tyk-operator-system generic tyk-operator-conf \
+  --from-literal "TYK_AUTH=foo" \
+  --from-literal "TYK_ORG=myorg" \
+```
+
+We can now reference our secret in the `OperatorContext` resource with `.spec.secretRef`
+
+```yaml
+apiVersion: tyk.tyk.io/v1alpha1
+kind: OperatorContext
+metadata:
+  name: community-edition
+spec:
+  secretRef:
+    namespace: tyk-operator-system
+    name: tyk-operator-conf
+  env:
+    mode: ce
+    url: http://tyk.tykce-control-plane.svc.cluster.local:8001
+    insecureSkipVerify: true
+    ingress:
+      httpPort: 8000
+      httpsPort: 8443
+```
+
+Mappings between `.spec.env` properties and secret `.spec.data` keys
+
+| secret key | .spec.env |
+|------------|-----------|
+| TYK_MODE   | mode      |
+| TYK_URL    | url       |
+| TYK_AUTH   | auth      |
+|TYK_ORG     | org       |
```

Comment:
lint: `| TYK_ORG     | org       |` instead of `|TYK_ORG     | org       |`

Reply:
done ac83456